configure

```diff
@@ -4049,6 +4079,31 @@ fi
 define_in_config_if_yes "$_opengl" "USE_OPENGL"
 define_in_config_if_yes "$_opengles" "USE_GLES"
 
+# Check if Raspberry Pi dispmanx has been activated. 
+if test "$_dispmanx" = "yes" ; then
+		echocheck "DispmanX graphics"
+		_use_dispmanx=no
+		DISPMANX_CXXFLAGS="-I$RPI_ROOTDIR/opt/vc/include -I$RPI_ROOTDIR/opt/vc/include/interface/vmcs_host/linux/ -I$RPI_ROOTDIR/opt/vc/include/interface/vcos/pthreads -mfpu=vfp -mfloat-abi=hard -I$RPI_ROOTDIR/opt/rpi_root/usr/include/SDL"
+		DISPMANX_LIBS="--sysroot=$RPI_ROOTDIR -L$RPI_ROOTDIR/usr/lib -L$RPI_ROOTDIR/opt/vc/lib -lbcm_host -lvcos -lvchiq_arm"
+		cat > $TMPC << EOF
+#include <bcm_host.h>
+
+		int main(int argc, char *argv[]) {
+	bcm_host_init();
+}
+EOF
+		cc_check $DISPMANX_CXXFLAGS $DISPMANX_LIBS && _use_dispmanx=yes
+		if test "$_use_dispmanx" = "yes"; then
+			CXXFLAGS="$CXXFLAGS $DISPMANX_CXXFLAGS"
+			LIBS="$LIBS $DISPMANX_LIBS"
+			MODULES="$MODULES backends/platform/sdl"
+			DEFINES="$DEFINES -DRASPBERRYPI"
+			add_line_to_config_mk 'RASPBERRYPI = 1'
+			_opengl=no
+			_opengles=no
```

Comment:
In fact, it seems that the result of this detection step affect which backend should be used -- if this check fails, your custom raspberrypi backend does not build. Yet you hard code that `_backend` value earlier (I already commented on that). Thus if dispmanx cannot be found, should't you either error out, or fallback to the regular SDL backend? Otherwise, the user will end up with a broken build environment with no visible hint that something is wrong, except for a tiny message that dispmanx was not found.